### PR TITLE
Week2_Day03_Inheritance_and_ClassTypes

### PR DESCRIPTION
**Pull request by Nayf-S-ALzahrani**
Week2_Day03_Inheritance_and_ClassTypes

### CODE REVIEW

File: src/homework2.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+}
+
+open class BaicShap() {// the prime class (the class will inherit the others)
+    var shapeName: String
+    var shapeColor: String
+
+    init {//initialization the variables
+        shapeName = ""
+        shapeColor = ""
```
Comment (shuruqt1000) [inline src/homework2.kt:25]: why did you initiate these values ?

File: src/homework2.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        pi = 3.14
+        radius = 5.1
+    }
+
+    override fun computeArea(): Double {
+        return (pi * (radius * radius))//arithmetic operation of the shape
+    }
+
+    override fun info(): String { //return shape information
+        shapeName= "Circle"
```
Comment (shuruqt1000) [inline src/homework2.kt:54]: why did you override this ?

**Review by shuruqt1000 — CHANGES_REQUESTED** [review]

File: src/homework2.kt (hunk 1 above)
Comment (Nayf-S-ALzahrani) [reply src/homework2.kt:25]: They are only initial values, and the values ​​will be assigned by the functions of the shapes.

File: src/homework2.kt (hunk 2 above)
Comment (Nayf-S-ALzahrani) [reply src/homework2.kt:54]: Because I'll give its values ​​inside each function.

**Comment by shuruqt1000** [comment]
completed

**Pull request closed without merging**